version: '3.8'

services:

  mlflow:
    build: ./docker/mlflow
    container_name: mlflow
    environment:
      MLFLOW_S3_ENDPOINT_URL: http://minio:9000
      AWS_ACCESS_KEY_ID: minioadmin
      AWS_SECRET_ACCESS_KEY: minioadmin
      MLFLOW_S3_IGNORE_TLS: "true"
    ports:
      - "8083:8083"
    depends_on:
      - minio
      - mysql
    restart: always
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8083/"]
      interval: 30s
      timeout: 10s
      retries: 3
  
  minio:
    container_name: Minio
    command: server /data --console-address ":8089"
    environment:
      - MINIO_ROOT_USER=minioadmin
      - MINIO_ROOT_PASSWORD=minioadmin
    image: quay.io/minio/minio:latest
    ports:
      - '8088:8088'
      - '8089:8089'
    volumes:
      - ./minio_data:/data
    restart: unless-stopped

  fastapi:
    build: ./docker/fastapi
    container_name: inference
    environment:
      MLFLOW_S3_ENDPOINT_URL: http://minio:9000
      AWS_ACCESS_KEY_ID: minioadmin
      AWS_SECRET_ACCESS_KEY: minioadmin
      MLFLOW_S3_IGNORE_TLS: "true"
      MLFLOW_TRACKING_URI: http://mlflow:5000
      DEFAULT_ARTIFACT_ROOT: s3://mlflows3/
    ports:
      - '8085:8085'
    restart: always
    networks:
      - network1
  
  networks:
    network1:
      name: p2_default
      external: true




